packages/react-router/src/ScriptOnce.tsx

```diff
@@ -12,6 +13,23 @@ export function ScriptOnce({
     return null
   }
 
+  // Validate input to prevent XSS
+  invariant(
+    typeof children === 'string',
+    'ScriptOnce children must be a string to prevent XSS attacks',
+  )
+
+  // Additional safety check for potentially dangerous content
+  if (
+    typeof process !== 'undefined' &&
+    process.env.NODE_ENV === 'development' &&
+    children.includes('<script')
```

Comment:
so a <script> inside a <script> is a problem? wouldn't any html tag be problematic?